### ISSUE TITLE
Disabling Share Presence also disables viewing read receipts and typing notifications

### ISSUE THREAD

**Issue by BurnyBoi** [SchildiChat/schildichat-android-next#31]
**Describe the bug**
Disabling Share Presence also causes other peoples read receipts and typing notifications to hide from the UI

**To Reproduce**
1.Check that Share Presence is enabled under Advanced Settings
2. Observe that read receipts and typing notifications display in a room
3. Disable Share Presence under Advanced Settings
4. Observe the same room and see that typing notifications and read receipts no longer display

**Expected behavior**
Read receipts and typing notifications from others should still display even if you choose to not share your own.

**Smartphone (please complete the following information):**
 - Device: Pixel 7
 - OS: Android 14

**Additional context**
 - App version: 0.7.2.sc13
 - Store: SpiritCroc's F-Droid
 - Homeserver: Self-hosted synapse 1.117.0

**Upstream relevance**
 - Does this issue exist in Element X Android? 
 Yes
 - If this issue exists in Element X Android, please also link the corresponding bug report (create a new one if it doesn't exist yet)
 https://github.com/element-hq/element-x-android/issues/3759

Element X devs are choosing to not separate viewing and sending read/typing receipts into different settings ["As a product decision"](https://github.com/element-hq/element-x-android/issues/3759). Element Web & Schildichat (not Next) on Android currently support not sending read receipts & typing notifs while still seeing others.

**Comment by disabledautomatic** [comment]
> Yes, this is working as expected and was a product decision.

Its wild to me how they just state this without any explanation for why. Voice calls between desktop and mobile are broken unless you use ElementX. All this while ElementX is still pretty much non functional with how many basic things it is missing. 

So yeah big +1 on this and thank you for making a fork of ElementX to make it usable.

**Comment by BurnyBoi** [comment]
My best guess is that New Vector doesn't want people "lurking" in chats undetected while being able to see everyone else's typing and read receipts. That being said, they currently allow this in Element Desktop, and being able to read the chat without anyone knowing you're there could be beneficial for moderators too.

**Comment by disabledautomatic** [comment]
New message/Notification badges are actually completely broken without read receipts because thats the only way your account can track your own read status across devices. So your options are "let people stalk your online status" or "live with a buggy mess". Thats just not a choice that a user should have to make. 

This is potentially privacy sensitive data that anyone could log by just observing a public room the user has joined, it absolutely has to be possible to disable this without completely losing basic multi device functionality. Maybe they just want to collect more metadata from users...

**Comment by SpiritCroc** [comment]
> New message/Notification badges are actually completely broken without read receipts because thats the only way your account can track your own read status across devices. So your options are "let people stalk your online status" or "live with a buggy mess". Thats just not a choice that a user should have to make.

It used to be like that but is no longer correct [since the introduction of private read receipts](https://spec.matrix.org/v1.12/client-server-api/#private-read-receipts), which one can use to clear notification count while not sharing anything with other chat members

**Comment by disabledautomatic** [comment]
So just to clarify this, from what [i can see](https://github.com/element-hq/element-meta/issues/2203#issuecomment-1926455742) all that happens when toggling "Share Presence" is that these 4 booleans get set. This still seems to be how it is implemented in the current build (from just quickly going through source with Ctrl+F, because i dont actually have any idea how Kotlin works)

- renderTypingNotifications
- sendTypingNotifications
- renderReadReceipts
- sendPublicReadReceipts

So to formulate a goal for what this maybe could/should look like...

In current non X Element we have 4 separate toggles for all of these which is (imo) optimal in terms of flexibility, but i can understand not wanting 4 toggles for this on the mobile app. Idk if you can do master/slave type toggles but that would be cool because it allows easy toggling for average users but also fine tuning for more advanced users.

I think the minimum should be 2 toggles:
- Read receipts send+receive
- Typing notifs send+receive

But there could be people that want to only receive but not send these events which also kind of has a place in a privacy oriented chat app imo.

So you could also conceive of two toggles like:
- Send Presence (Read receipts send + Typing notifs send)
- Receive Presence (Read receipts receive + Typing notifs receive)

What are your thoughts?

**Comment by BurnyBoi** [comment]
I agree. 4 distinct actions. Currently one toggle controls all, and some other clients will do the two toggles like you mentioned. Ideally we should have two.
